tutor/src/models/student-tasks/task.js

```diff
@@ -57,14 +57,14 @@ class StudentTask extends BaseModel {
     if (this.isLoading) return this;
     this.isLoading = true;
     try {
-      let tries_remaining = 30;
+      let tries_remaining = 6;
       while(tries_remaining > 0) {
         await this.fetch();
         if (this.isLoaded) {
           break;
         }
-        // wait 1 second in between load attempts
-        await new Promise(resolve => setTimeout(resolve, 1000));
+        // exponential backoff between tries: 1, 2, 4 .. 32 (plus some milliseconds of random)
+        await new Promise(resolve => setTimeout(resolve, 1000*(2**(6 - tries_remaining) + Math.random()));
         tries_remaining--;
```

Comment:
Nice!  But.. `Math.random()` returns a float from 0 - 1 so that will either round to 0 or 1.  Probably need to `(Math.random() * 100)`

Reply:
that's what I wanted, I do the calc in seconds then multiply the whole thing by 1000.  (but failed to get all the nested parens back in place. grrr)

Reply:
(guess I need to learn how to run the nodejs spec test suite locally … *blush*

Reply:
Ahh rounding - I worried about that, but tried it at a node prompt, and got a float result.

Reply:
> tries_remaining=6
6
> (2**(6 - tries_remaining) + Math.random())*1000
1748.102522071031
> (2**(6 - tries_remaining) + Math.random())*1000
1343.0246008299464
>